cmd/process-agent/main.go

```diff
@@ -8,14 +8,20 @@
 package main
 
 import (
+	"context"
+	"os"
+
 	"github.com/DataDog/datadog-agent/cmd/process-agent/command"
 )
 
 const useWinParams = false
 
 func rootCmdRun(globalParams *command.GlobalParams) {
-	exit := make(chan struct{})
-
 	// Invoke the Agent
-	runAgent(globalParams, exit)
+	err := runAgent(context.Background(), globalParams)
+	if err != nil {
+		// For compatibility with the previous cleanupAndExitHandler implementation, os.Exit() on error.
+		// This prevents runcmd.Run() from displaying the error.
```

Comment:
its removal is defferred in [runAgent](https://github.com/DataDog/datadog-agent/pull/18702/files#diff-ceb61f735a2b8dbd182ee59e99b933a6fa931f3ea10c7730b287f47b7021a5b1L83-L88). `cleanupAndExitHandler` was necessary before because `runAgent` never returned, so the defer was never executed. But now that `runAgent` returns we can rely on the `defer` to cleanup the pid file.